File: Score Counter Animated - Example Project/Score Counter Animated/Graph/ScriptTemplate/Floor/CGFloor.js
/**
 * @file CGFloor.js
 * @author liujiacheng
 * @date 2021/8/23
 * @brief CGFloor.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGFloor extends BaseNode {
  constructor() {
    super();
  }

  setNext(index, func) {
    this.nexts[index] = func;
  }

  setInput(index, func) {
    this.inputs[index] = func;
  }

  getOutput() {
    return Math.floor(this.inputs[0]());
  }
}

exports.CGFloor = CGFloor;

File: Score Counter Animated - Example Project/Score Counter Animated/Graph/ScriptTemplate/Floor/CGFloor.lua
local CGFloor = CGFloor or {}
CGFloor.__index = CGFloor

function CGFloor.new()
    local self = setmetatable({}, CGFloor)
    self.inputs = {}
    return self
end

function CGFloor:setInput(index, func)
    self.inputs[index] = func
end

function CGFloor:getOutput(index)
    return math.floor(self.inputs[0]())
end

return CGFloor
